### ISSUE TITLE
windows-specific code?

### ISSUE THREAD

**Issue by davebarr** [TheSylex/ELK-BLEDOM-bluetooth-led-strip-controller#1]
I'm new to rust as well as btle.  It appears that this program uses windows-specific APIs (struct) when defining BleLedDevice:

```rust
pub struct BleLedDevice {
    peripheral: btleplug::winrtble::peripheral::Peripheral,
    characteristics: Vec<btleplug::api::Characteristic>,
}
```
when building on linux I get:

```
% cargo build
   Compiling blte_test v0.1.0 (/home/pi/git/ELK-BLEDOM-bluetooth-led-strip-controller)
error[E0433]: failed to resolve: could not find `winrtble` in `btleplug`
   --> src/device.rs:106:27
    |
106 |     peripheral: btleplug::winrtble::peripheral::Peripheral,
    |                           ^^^^^^^^ could not find `winrtble` in `btleplug`
```

I'm digging through the code trying to refactor this but not having much luck, the btleplug API is a bit of a mess.

**Comment by davebarr** [comment]
Ah there's a long explanation about this in https://github.com/deviceplug/btleplug/issues/73, with not a simple answer, alas

**Comment by TheSylex** [comment]
Yeah, actually I tried to do it with the generalized struct but couldn't get it to work, I'm using it on Windows so it didn't bother me enough to dig deeper, so I just stick with the windows one. Feel free to do a pull request or suggest a change if you find the right way.

**Comment by Aex12** [comment]
Same problem here, not compiling on linux :(

**Comment by Michas0206** [comment]
Hi, it would be so cool if this would run on a Raspberry Pi / Linux...with animation sync to soundoutput...or as a Kodi addon....unfortunately I dont know anything about Rust.

**Comment by legop3** [comment]
I am on linux trying to figure this out right now, this is the first time ive ever touched rust.
